### DISCUSSION TITLE
Hi, can this asset help me with Google play pass?

### DISCUSSION THREAD

**Discussion by pavemi95** [KirillKuzyk/Google-Play-License-Check#16] [Q&A] [answered]
Hi I want to buy this asset, but I’m not sure this is the right one for me. I need an asset to help me build a paywall from Play when a subscription is no longer active. I’m not a programming expert, can you tell me if this can help me? And if it can, what do I have to do? Is there a guide I can follow? Thank you so much!

**Comment by KyryloKuzyk** [comment, marked as answer]
Yes, if your app is paid, the plugin can help with Play Pass integration.

But if you have a free app, the Google Play License Check plugin can't help check the Play Pass subscription status. The recommended way of doing it for free apps is by implementing an in-app purchase, as described here:
https://github.com/KirillKuzyk/Google-Play-License-Check/discussions/6#discussioncomment-3314455

For free games, the plugin can validate that your game was downloaded from Google Play and verify the app's integrity. But to check the subscription status, you have to use another plugin.

**Reply by pavemi95** [reply]
No but mine is a paid game. Can this help me to set the google play pass subscription paywall? Thank you. Sorry if I'd speak badly, I'm not native speaker

**Reply by KyryloKuzyk** [reply]
If you have a paid game, then my plugin perfectly suits for making a Play Pass paywall.
In the case of paid apps, it’s not necessary to check the Play Pass subscription status. The official way in this case is to use the Licensing Service and check if license status is LICENSED (and the grace period is not elapsed). All of this is supported by my plugin.

**Reply by pavemi95** [reply]
Hi, thank you for your reply. I followed every step in your documentation. Does it mean that my game is ready for play pass? Because I tried to simulate a not licensed version on unity and the game worked anyhow.

And can I make it possible that the check of the license is done every day? Not everyone the user opens the app? Thank you!

**Reply by KyryloKuzyk** [reply]
If you've integrated my plugin into your app and followed the Play Pass guidelines, then yes, your app should be ready. Please let me remind you of the key guidelines of Play Pass:
1. If the license status is NOT_LICENSED, then the app should present the Play Pass paywall. Here is an example:
```
GooglePlayLicense.Check(result => {
    if (result.status == LicenseStatus.NOT_LICENSED) {
        var licensingUrl = result.licensingUrl;
        if (licensingUrl != null) {
            GooglePlayLicense.OpenLicensingUrl(licensingUrl);
        } else {
            Application.OpenURL($"market://details?id={Application.identifier}");
        }
        Application.Quit(); // todo decide if quitting is the behavior that suits your app
    }
});
```
2. Play Pass recommends giving users 30 days grace period. Here is an example:
```
public static bool IsUserEligibleForUsingTheApp() {
    var cached = GooglePlayLicense.cachedLicense;
    if (cached == null) {
        return false;
    }
    // Google Play Pass recommends to cache the license response for 30 days.
    return cached.status.IsLicensed() && DateTime.Now < cached.timestamp + TimeSpan.FromDays(30);
}
```

> Because I tried to simulate a not licensed version on unity and the game worked anyhow.

Simulated license responses are meant only for debugging. I would strongly suggest you to test your app on real devices as described here before releasing it to production:
https://github.com/KirillKuzyk/Google-Play-License-Check#closedopen-testing-track



> And can I make it possible that the check of the license is done every day? Not everyone the user opens the app? Thank you!

Yes, it's possible, but you'll have to implement it yourself. The plugin does the license check every time you call GooglePlayLicense.Check(), so you should call this method only when you feel it's right for your use case. Although, I see no strong reasons why the license check should not be performed every app launch.
